### PR TITLE
Merging joshuarubin master

### PR DESCRIPTION
**Pull request by meatballhat**

### CODE REVIEW

File: flag.go [hunk 1]
```diff
@@ -578,8 +578,16 @@ func withEnvHint(envVar, str string) string {
 	return str + envText
 }
 
-func stringifyFlag(f Flag) string {
+func flagValue(f Flag) reflect.Value {
 	fv := reflect.ValueOf(f)
+	for fv.Kind() == reflect.Ptr {
+		fv = reflect.Indirect(fv)
+	}
+	return fv
```
Comment (joshuarubin) [inline flag.go]: no need to test for the pointer, `Indirect` returns the `Value` if it is not a pointer. You can just do:

```go
return reflect.Indirect(reflect.ValueOf(f))
```

Comment (meatballhat) [inline flag.go]: Ack!  Pointers to pointers aren't covered, AFAICT, although that's getting pretty edgy.  Maybe compare to the zero value instead?  (ref https://golang.org/src/reflect/value.go#L2075)

Comment (joshuarubin) [inline flag.go]: ha, didn't see that it was a `for`, presumed it was an `if`.

Comment (meatballhat) [inline flag.go]: Whaddya think, then?  Leave as-is?

Comment (joshuarubin) [inline flag.go]: Yeah, couldn't hurt.

**Pull request merged**